pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -551,33 +553,40 @@ func (c *Cluster) initMonIPs(mons []*monConfig) error {
 	return nil
 }
 
-func (c *Cluster) assignMons(mons []*monConfig) error {
-	// when monitors are scheduling below by invoking scheduleMonitor() a canary
-	// deployment and optional canary PVC are created. in order for the
-	// anti-affinity rules to be effective, we leave the canary pods in place
-	// until all of the canaries have been scheduled. only after the
-	// monitor/node assignment process is complete are the canary deployments
-	// and pvcs removed here.
-	canaryCleanup := []SchedulingResult{}
-	defer func() {
-		for _, result := range canaryCleanup {
-			logger.Infof("assignmon: cleaning up canary deployment %s and canary pvc %s", result.CanaryDeployment, result.CanaryPVC)
-			if result.CanaryDeployment != "" {
-				if err := k8sutil.DeleteDeployment(c.context.Clientset, c.Namespace, result.CanaryDeployment); err != nil {
-					logger.Infof("assignmon: error deleting canary deployment %s. %v", result.CanaryDeployment, err)
-				}
+// Delete mon canary deployments (and associated PVCs) using deployment labels
+// to select this kind of temporary deployments
+func (c *Cluster) removeCanaryDeployments() {
+	canaryDeployments, err := k8sutil.GetDeployments(c.context.Clientset, c.Namespace, "app=rook-ceph-mon, mon_type=canary")
```

Comment:
I have never seen an example of a label selector in the k8s docs with a space after the comma. I'm not sure this will work.
```suggestion
	canaryDeployments, err := k8sutil.GetDeployments(c.context.Clientset, c.Namespace, "app=rook-ceph-mon,mon_type=canary")
```

Reply:
Removed the extra blank space. And, yes...  it works , at least the mon canary deployments are selected properly.:-)